src/xref2/tools.ml

```diff
@@ -1579,7 +1579,7 @@ and expansion_of_module_path :
           let sg' =
             match m.doc with
             | [] -> sg
-            | docs -> { sg with items = Comment (`Docs docs) :: sg.items }
+            | doc -> { sg with doc }
```

Comment:
I'm still really not familiar with this part of the code...
Could this erase some content from `sg`? (Could `{ sg with doc = doc @ sg.doc }` be safer?)
And why is this done only in `expansion_of_module_path`, and not in `expansion_of_module`, since this function already has `m`?

Reply:
It's overriding the doc for this use case:
```
module P = struct
  (** Preamble for P. *)
end

(** Outside preamble for Q. *)
module Q = struct
  (** Inside preamble for Q. *)
end
```

We want to see the preamble from the first comment in `P` but we want the comment outside of `Q`. Concatenating them would result in the same bug this patch is trying to fix.

Moving that to `expansion_of_module` makes complete sense.

Reply:
Thanks for the explanation. This makes sense!